csbuild/toolchain/__init__.py

```diff
@@ -163,6 +174,38 @@ def RunGroup(self, inputProject, inputFiles):
 			raise NotImplementedError()
 		return ""
 
+class SolutionGenerator(object):
+	"""
+	Solution Generator base class.
+
+	Shares a similarity with tools in that it has a list of input files (in this case, projectExtensions),
+	but otherwise, is little more than a wrapper around the GenerateSolution() static method, which will
+	be called in an OnBuildFinished hook. Making it a class allows extension via additional
+	static data and methods.
+	"""
+
+	#: List of project file extensions created by the project generator tool.
+	#  Alternatively, a solution generator could be set up without a project generator tool dependency
+	#  by listing raw source file extensions instead.
+	#  Example: Visual Studio project generator would take {".proj"}
```

Comment:
Not to be overly pedantic, but VS doesn't use ".proj" files.  Each project actually has a collection of different file types associated with it.

I think it might be better to word this in a more generic way.  The way I plan to use it is to have the project generators collect the info I need for each project per target/platform/arch/etc and save that data to a staging json file, then the solution generator would handle merging the data and creating all the real project files.